### ISSUE TITLE
Fix f_mntonname derivation in fstatfs64.

### ISSUE THREAD

**Issue by r-wilbur** [darlinghq/darling#946]
Using fstatfs64 I expected to get the same results as statfs64 on the same file.

fstatfs64 reported f_mntonname == "/" where statfs64 reported f_mntonname == "/home/<uname>/.darling" on the same file.

**Steps To Reproduce**
1. Call statfs64 on a filename of a filesystem mounted on "/home/<uname>/.darling" and print the f_mntonname field.
2. Open the same file and pass the file descriptor to fstatfs64.  Print the f_mntonname field returned.

**System Information**
Linux 5.4.0-66-generic SMP x86_64

| Software | Version |
| --- | --- |
| Linux Kernel | X.Y.Z | 5.4.0
| Darling | Git Commit Hash | 1d3b09970856d58d12867f4ee5b9b1bf84cd05c2

**Comment by facekapow** [comment]
Actually, I think `statfs64` is the one that's broken here: it's returning paths that aren't vchrooted. We need to be `vchroot_unexpand`-ing the path we assign to `f_mntonname`.

**Comment by r-wilbur** [comment]
Now that I look at it from the perspective of the emulated environment, I see you are correct.  For the purposes of the extended attribute code I was working on, I needed to determine the provenance of the underlying filesystem:  specifically was it provided by darling (FUSE and mounted on /home/<userid>/.darling) or provided by linux.  Thus I inadvertently uncovered a bug which was useful to me.  Sounds like I need to make a function to reproduce the incorrect behavior of statfs64 in the extended attribute code and then fix statfs64().
@facekapow Thanks for the correction!

**Comment by r-wilbur** [comment]
It was broken in both fstatfs64 and statfs64 in the following way:
statfs64:  expanded_path "/home/awilbur/.darling/Volumes/macOS Mojave Patcher/macOS Mojave Patcher.app/Contents"
statfs64: line "darling-dmg /home/awilbur/.darling/Volumes/macOS\040Mojave\040Patcher fuse.darling-dmg ro,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0"
expanded path was expanded by vchroot correctly and used to make the Linux system call and then to compare with lines from /proc/mounts.
The line which should have been the best match shows up with " " becoming "\040" and thus doesn't match in strcmp().  Looks like I need to un-escape the string read from /proc/mounts before comparing it to the expanded path.  Then after all is said and done, we should vchroot_unexpand the f_mntonname.

Do we have a handy function to turn those octal character codes back into normal characters in a string?

**Comment by LubosD** [comment]
> Do we have a handy function to turn those octal character codes back into normal characters in a string?

No, I don't think there's such a function in the limited environment of libsystem_kernel.

**Comment by r-wilbur** [comment]
Turns out it was important to have those octal character codes in paths in /proc/mounts in order to be able to tokenize and split the line on " " (space) characters.  But after we have split the line, it is important to achieve a uniform representation to simplify matching paths from the filesystem via strcmp and the like.  So I implemented a routine to "un-escape" and normalize the names to facilitate comparisons for both fstatfs64 and statfs64.  The normalization involves simplifying multiple forward slashes into a single forward slash '/'.
